### PR TITLE
fix: sip10 tokens send form bug

### LINKED ISSUE #3537: Clicking certain SIP010 tokens to send results in going back to the main wallet screen
On The Wallet Extension with software secret key:

https://user-images.githubusercontent.com/33360391/231159778-59a57bcf-09b4-4b99-8792-1a676ee0fc24.mp4

Contracts of failing tokens:
wrapped nothing v8: https://explorer.hiro.so/txid/SP32AEEF6WW5Y0NMJ1S8SBSZDAY8R5J32NBZFPKKZ.wrapped-nothing-v8?chain=mainnet
friedger: https://explorer.hiro.so/txid/SPN4Y5QPGQA8882ZXW90ADC2DHYXMSTN8VAR8C3X.friedger-token-v1?chain=mainnet

Contract of working token (MEGA)... going to the send form. 
https://explorer.hiro.so/txid/SP3D6PV2ACBPEKYJTCMH7HEN02KP87QSP8KTEH335.mega?chain=mainnet

On Ledger enabled wallets this results in: https://github.com/hirosystems/wallet/issues/3536

Does this have anything to do with (un)available metadata?

### PR DESCRIPTION
**Pull request by alter-eggo**
> Try out this version of the Hiro Wallet - download [extension builds](https://github.com/hirosystems/wallet/actions/runs/4685804828).

Closes #3537 

There is also a problem with decimals in sent tx summary, it is fixed in other pr - https://github.com/hirosystems/wallet/pull/3494

### CODE REVIEW

**Comment by 314159265359879** [comment]
LGTM I can send wrapped nothing v8 now and friedger

File: src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx [hunk 1]
```diff
@@ -36,23 +36,29 @@ interface AmountFieldProps {
...
 
-  const { decimals, symbol } = balance;
+  const { decimals } = balance;
+  const symbol = tokenSymbol || balance.symbol;
   const maxLength = decimals === 0 ? maxLengthDefault : decimals + 2;
   const fontSizeModifier = (maxFontSize - minFontSize) / maxLength;
 
   useEffect(() => {
+    if (symbol.length > 4) {
+      setFontSize(24);
```
Comment (fbwoolf) [inline src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx]: Might read better if this was done on one line, but not critical. Curious the significance of `24`, should that be set as a named constant?

**Review by fbwoolf — APPROVED** [review]

File: src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx (hunk 1 above)
Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx]: yeah, I changed to `minFontSize` constant

File: src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx [hunk 2]
```diff
@@ -36,23 +36,29 @@ interface AmountFieldProps {
...
   autoComplete = 'on',
+  tokenSymbol,
 }: AmountFieldProps) {
   const [field, meta] = useField('amount');
   const [fontSize, setFontSize] = useState(maxFontSize);
   const [previousTextLength, setPreviousTextLength] = useState(1);
 
-  const { decimals, symbol } = balance;
+  const { decimals } = balance;
+  const symbol = tokenSymbol || balance.symbol;
```
Comment (kyranjamie) [inline src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx:54]: When would these not be the same?

**Review by kyranjamie — APPROVED** [review]

File: src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx (hunk 2 above)
Comment (alter-eggo) [reply src/app/pages/send/send-crypto-asset-form/components/amount-field.tsx:54]: when, e.g. sip10 token contract doesn't have symbol

**Pull request merged**